### PR TITLE
bootloader: pm.yml: Move provision into b0 region

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Since it needs to be protected, this works better in both nrf91 and
nrf5x.

The nrf91 part will eventually be moved to the OTP instead.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

### CODE REVIEW

**Comment by oyvindronningstad** [comment]
cc @sigvartmh @osaether

File: subsys/bootloader/Kconfig [hunk 1]
```diff
@@ -122,7 +122,7 @@ menuconfig IS_SECURE_BOOTLOADER
 if IS_SECURE_BOOTLOADER
 
 partition=B0
-partition-size=0x8000
+partition-size=0x7000
```
Comment (sigvartmh) [inline subsys/bootloader/Kconfig]: Why is this changed? isn't it `b0_image` which should be 0x7000? Since `b0` spans `image + provision data`? It could be that I don't understand partition manager correctly.

Comment (hakonfam) [reply subsys/bootloader/Kconfig]: Hm, as far as I remember it must reflect the name in pm.yml

Comment (oyvindronningstad) [reply subsys/bootloader/Kconfig]: Yep, forgot to update the name, will fix.

Comment (oyvindronningstad) [reply subsys/bootloader/Kconfig]: done

**Comment by oyvindronningstad** [comment]
> IMO "The nrf91 part will eventually be moved to the OTP instead." should not be in the commit message, but a JIRA task instead.

Removed the line

File: subsys/bootloader/Kconfig [hunk 2]
```diff
@@ -121,8 +121,8 @@ menuconfig IS_SECURE_BOOTLOADER
 
 if IS_SECURE_BOOTLOADER
 
-partition=B0
-partition-size=0x8000
+partition=B0_IMAGE
```
Comment (hakonfam) [inline subsys/bootloader/Kconfig]: should this be lower case? I see now that that is the case in the other Kconfig files.

Comment (oyvindronningstad) [reply subsys/bootloader/Kconfig]: IN MCUboot they are upper case, so I think this line is currently following convention. We can change it later.

Comment (sigvartmh) [reply subsys/bootloader/Kconfig]: `B0_image` middle ground?

Comment (oyvindronningstad) [reply subsys/bootloader/Kconfig]: I think we should leave it as is for now.

Comment (hakonfam) [reply subsys/bootloader/Kconfig]: I agree that we follow what is there, just me mis-remembering :)

File: samples/bootloader/pm.yml [hunk 3]
```diff
@@ -1,12 +1,15 @@
 #include <autoconf.h>
 #include <generated_dts_board_unfixed.h>
 
-b0:
-  size: CONFIG_PM_PARTITION_SIZE_B0
+b0_image:
+  size: CONFIG_PM_PARTITION_SIZE_B0_IMAGE
   placement:
     after: start
     align: {end: 0x8000}
```
Comment (hakonfam) [inline samples/bootloader/pm.yml]: While you are at it, you can remove this alignment. Since this is the first image in flash, we can not align it :) Also the end alignment does not makes ense when the provision is behind it. It should be the proviison partition that had the alignment requriment.

Comment (oyvindronningstad) [reply samples/bootloader/pm.yml]: Done
